### ISSUE TITLE
[FEATURE] NUCAPS?

### ISSUE THREAD

**Issue by rpdale** [sharppy/SHARPpy#222]
I've seen mention of NUCAPS and SHARPpy, but can't find a source... Anyone know of a secret URL :) ? The CIMSS blog talks about a website source but I'm coming up empty.

https://cimss.ssec.wisc.edu/satellite-blog/archives/42650

**Comment by Jeff-Szkodzinski** [comment]
Hello rpdale!

I'm glad to hear that you're interested in working with NUCAPS data in SHARPpy!  We're in the process of migrating our NUCAPS-specific enhancements for SHARPpy to the master branch so that all the code can be downloaded from one central location.  There is another branch called "nucaps-test" that we've been adding enhancements/fixes to before we eventually merge the code to master and it becomes part of the baseline.  If you're trying to install SHARPpy to your machine for the first time, you can run "git clone -b nucaps-test --single-branch https://github.com/sharppy/SHARPpy/tree/nucaps-test.git" in a terminal to get the NUCAPS data sources.  In the future when we push additional changes to the nucaps-test repository, you can download the latest updates by typing "git pull origin nucaps-test".  Let me know if you have any trouble downloading from the nucaps-test branch or if the code behaves funny in any way and I can take a look into it.

Thanks!
Jeff

**Comment by Jeff-Szkodzinski** [comment]
Try "git clone -b nucaps-test --single-branch https://github.com/sharppy/SHARPpy.git" if the above doesn't work.

**Comment by rpdale** [comment]
Awesome - thanks! I was able to install using the second line and most of the program works...

![1](https://user-images.githubusercontent.com/11564694/144712658-a888452b-1c6f-40e5-ad29-471839e2a002.gif)
Everything except NUCAPS :)


![2](https://user-images.githubusercontent.com/11564694/144712663-8cb233ca-5586-4c4f-bf44-2d2efc096b37.gif)

When I select any of those options, it's updating the times available but not showing any dots to select.

Side notes: I get the popup about a new version being available at startup, and hitting Preferences does nothing except cause a bug report whether from the main menu or profile screen.

![3](https://user-images.githubusercontent.com/11564694/144712672-d449f149-be3c-4bbc-928b-544fa1ef0547.gif)

I'm running Anaconda on Win11.

**Comment by skovic** [comment]
@rpdale The preferences problem is a known issue (https://github.com/sharppy/SHARPpy/issues/224). I have a pull request ( https://github.com/sharppy/SHARPpy/pull/225) out to fix it. You could try to cherry pick the commit or manually make the change by hand in your code.

**Comment by Jeff-Szkodzinski** [comment]
@rpdale Were you able to see any NUCAPS points when you chose a specific time in the drop-down menu?  You'll need to select a satellite overpass time first (i.e., 0833 or 1016 UTC) before the points will display on the map.

**Comment by rpdale** [comment]
> @rpdale Were you able to see any NUCAPS points when you chose a specific time in the drop-down menu? 

Thanks for responding... But no. I've tried every time option available today and previous days, and all the different NUCAPS sources, and no joy. It flashes "Loading..." in the lower left corner for just a very split second, whereas the other models have that up for a few seconds before they show me points.

I tried accessing the NUCAPS URL directly from my browser and I see the txt files there just fine, and obviously it's seeing something in SHARPpy since it gets the timestamp, but just not displaying any points.

<img width="933" alt="image" src="https://user-images.githubusercontent.com/11564694/144727592-59a0e38a-df5c-4d32-a8e5-914ca66942a4.png">

Here's About:

<img width="496" alt="image" src="https://user-images.githubusercontent.com/11564694/144727680-8cd8bdcc-3a42-4be7-9997-159be51a8766.png">

**Comment by rpdale** [comment]
This may be too basic of a question to answer and if so I'll await the formal merge :) but is there a way I can grab your new NUCAPS version to try before that merge occurs?

**Comment by Jeff-Szkodzinski** [comment]
Absolutely!  You can try downloading SHARPpy from my master branch at https://github.com/Jeff-Szkodzinski/SHARPpy to see if you have better luck.  I wasn't able to reproduce the issue you're seeing when I tried cloning from nucaps-test, but I let the original developers know of the problem you're having to see if they have any ideas.

**Comment by rpdale** [comment]
Successful with your master branch!

<img width="1631" alt="image" src="https://user-images.githubusercontent.com/11564694/145112147-6311c8e5-6391-415b-84f8-bdb7a385c250.png">

**Comment by rpdale** [comment]
Reopened :) The only thing it doesn't seem to do is display the points until I select a different time -- so when I pick a model, it shows me times for the default timestamp, where this will keep the map blank until I pulldown and pick a different time in the bar. For example, right now 2040Z is the latest CONUS NOAA20 data, but I have to select 1855Z data and then select 2040Z to show it. But when I do that, SHARPPY crashes with QThread: Destroyed while thread is still running.

**Comment by Jeff-Szkodzinski** [comment]
I was able to reproduce the issue on my local Windows machine.  The program crash and QThread error seem to occur randomly for me when I click on different satellite overpasses from the drop-down menu.  I'm not sure if this is related directly to the code or a potential server-side issue.  I'll do some investigating and get back to you once I know more.

Thanks,
Jeff

**Comment by rpdale** [comment]
I'm glad it wasn't just me :) Thanks for the followup. The core is working fine though so I'm in much better shape than before!

**Comment by rpdale** [comment]
Hey Jeff - just touching base. I'm still getting the issue where I can't select the most current time. It defaults to that but shows no points, and I can pick an older time to see data. But then when I go back to current, I get the QThread crash.

**Comment by Jeff-Szkodzinski** [comment]
Hi rpdale,

Thanks very much for checking back in.  Unfortunately I haven't been able to look into the issue in that much depth as I've been on another project over the last several months.  I'd be happy to refer you to one of our other NUCAPS developers on the JPSS program, Rebekah Esmaili (rebekah@stcnet.com), who maintains our NUCAPS code for SHARPpy and would be able to do more investigation on her end.  Feel free to reach out to her with any questions or issues running the program.  

Best,
Jeff
